app/assets/javascripts/discourse/app/lib/sidebar/user/categories-section/category-section-link.js

```diff
@@ -7,6 +7,22 @@ import { bind } from "discourse-common/utils/decorators";
 import Category from "discourse/models/category";
 import { UNREAD_LIST_DESTINATION } from "discourse/controllers/preferences/sidebar";
 
+const UNREAD_AND_NEW_COUNTABLE = {
+  propertyName: "unreadAndNewCount",
+  badgeTextFunction: (count) => count.toString(),
+  route: "discovery.newCategory",
+  refreshCountFunction: ({ topicTrackingState, category }) => {
+    return (
+      topicTrackingState.countNew({
+        categoryId: category.id,
+      }) +
+      topicTrackingState.countUnread({
+        categoryId: category.id,
+      })
```

Comment:
I wonder if we can more efficient with the count here. Looking up the count requires us to loop through all the records in `TopicTrackingState` each time. Instead, I think we can update `TopicTrackingState` to expose a `countNewAndUnread` so that it returns the count in a single loop.

Reply:
This is done ✅